### PR TITLE
feat: slotDefault to children

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: core-bootstrap/src/components/accordion/accordion.ts [hunk 1]
```diff
@@ -28,24 +28,25 @@ interface AccordionExtraProps {
...
+	 * The itemTransition it applied on this element.
+	 */
+	structure: SlotContent<AccordionItemContext>;
 	/**
 	 * Content present in the accordion body.
-	 *
-	 * It is a prop of the accordion-item.
 	 */
-	slotItemBody: SlotContent<AccordionItemContext>;
+	children: SlotContent<AccordionItemContext>;
```
Comment (andreascorti) [inline core-bootstrap/src/components/accordion/accordion.ts:45]: has it the same meaning as other "children"?

File: core-bootstrap/src/components/pagination/pagination.spec.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ describe(`Pagination`, () => {
 	});
 
 	test('should include a simple slot number label factory in default config', () => {
-		const slotNumberLabel = getPaginationDefaultConfig().slotNumberLabel as (props: {displayedPage: number}) => string;
+		const slotNumberLabel = getPaginationDefaultConfig().numberLabel as (props: {displayedPage: number}) => string;
```
Comment (andreascorti) [inline core-bootstrap/src/components/pagination/pagination.spec.ts:72]:
```suggestion
		const numberLabel = getPaginationDefaultConfig().numberLabel as (props: {displayedPage: number}) => string;
```

File: demo/src/routes/docs/[framework]/components/pagination/playground/+page.svelte [hunk 3]
```diff
@@ -8,7 +8,7 @@
 		onPageChange: ['noop', 'log'],
 		pagesFactory: ['defaultPageFactory'],
 		ariaPageLabel: ['simpleLabel', 'pageLabel'],
-		slotNumberLabel: ['slotNumberLabel'],
+		numberLabel: ['slotNumberLabel'],
```
Comment (andreascorti) [inline demo/src/routes/docs/[framework]/components/pagination/playground/+page.svelte:11]:
```suggestion
		numberLabel: ['numberLabel'],
```

File: angular/bootstrap/src/components/pagination/pagination.component.ts [hunk 4]
```diff
@@ -188,7 +188,7 @@ export const paginationDefaultSlotPages = new ComponentTemplate(PaginationDefaul
 export const paginationDefaultSlotStructure = new ComponentTemplate(PaginationDefaultSlotsComponent, 'structure');
 
 const defaultConfig: Partial<PaginationProps> = {
-	slotStructure: paginationDefaultSlotStructure,
+	structure: paginationDefaultSlotStructure,
 	slotPages: paginationDefaultSlotPages,
```
Comment (andreascorti) [inline angular/bootstrap/src/components/pagination/pagination.component.ts:192]: Why is `slotPages` not migrated to `pages` ?

File: angular/bootstrap/src/components/rating/rating.component.ts [hunk 5]
```diff
@@ -107,6 +107,9 @@ export class RatingComponent extends BaseWidgetDirective<RatingWidget> implement
 	 */
 	@Input({alias: 'auResettable', transform: auBooleanAttribute}) resettable: boolean | undefined;
 
+	/**
+	 * The template to override the way each star is displayed.
+	 */
 	@Input('auSlotStar') slotStar: SlotContent<AdaptSlotContentProps<StarContext>>;
```
Comment (andreascorti) [inline angular/bootstrap/src/components/rating/rating.component.ts:113]: shouldn't be only `star` ?

File: core-bootstrap/src/components/select/select.ts [hunk 6]
```diff
@@ -44,7 +44,7 @@ export interface SelectProps<Item> extends CoreProps<Item>, SelectExtraProps<Ite
 export type SelectWidget<Item> = Widget<SelectProps<Item>, SelectState<Item>, SelectApi<Item>, SelectActions<Item>, SelectDirectives<Item>>;
 
 const defaultConfigExtraProps: SelectExtraProps<any> = {
-	slotBadgeLabel: ({itemContext}: SelectItemContext<any>) => itemContext.item,
+	badgeLabel: ({itemContext}: SelectItemContext<any>) => itemContext.item,
 	slotItem: ({itemContext}: SelectItemContext<any>) => itemContext.item,
```
Comment (andreascorti) [inline core-bootstrap/src/components/select/select.ts:48]: shouldn't be only `item`?

File: react/bootstrap/src/components/pagination/pagination.tsx [hunk 7]
```diff
@@ -50,7 +50,7 @@ export function DefaultPages(slotContext: PaginationContext) {
 
 const defaultConfig: Partial<PaginationProps> = {
 	slotPages: DefaultPages,
```
Comment (andreascorti) [inline react/bootstrap/src/components/pagination/pagination.tsx:52]: already remarked in core files

File: react/bootstrap/src/components/select/select.tsx [hunk 8]
```diff
@@ -64,7 +64,7 @@ function Rows<Item>({slotContext}: {slotContext: SelectContext<Item>; menuId: st
 }
 
 const defaultConfig: Partial<SelectProps<any>> = {
-	slotBadgeLabel: DefaultBadge,
+	badgeLabel: DefaultBadge,
 	slotItem: DefaultItem,
```
Comment (andreascorti) [inline react/bootstrap/src/components/select/select.tsx:68]: remarked in core files

File: svelte/bootstrap/src/components/modal/Modal.svelte [hunk 9]
```diff
@@ -8,15 +8,15 @@
...
+		structure: ModalDefaultStructure,
+		header: ModalDefaultHeader,
 	};
 </script>
 
 <script lang="ts">
 	type Data = $$Generic; // eslint-disable-line no-undef
 	type $$Props = Partial<ModalProps<Data>>;
-	type $$Slots = ModalSlots<Data>;
+	type $$Slots = ModalSlots<any>;
```
Comment (andreascorti) [inline svelte/bootstrap/src/components/modal/Modal.svelte:19]: why did you have to change to `any`?

File: svelte/bootstrap/src/components/rating/Rating.svelte [hunk 10]
```diff
@@ -36,7 +36,7 @@
 		<span class="visually-hidden">({index < $visibleRating$ ? '*' : ' '})</span>
 		<span use:starDirective={{index}}>
 			<Slot slotContent={$slotStar$} props={{fill, index}} let:component let:props>
-				<svelte:fragment slot="slot" let:props><slot name="star" {...props} /></svelte:fragment>
+				<svelte:fragment slot="slot" let:props><slot name="slotStar" {...props} /></svelte:fragment>
```
Comment (andreascorti) [inline svelte/bootstrap/src/components/rating/Rating.svelte:39]: Why ?

**Review by andreascorti — COMMENTED** [review]
Thanks @quentinderoubaix for the refactor. 
Some comments and general remark:

In pagination, is it on purpose you left `slotPages`? 
Same for Rating, `slotStar` and select `slotItem`.

Also, not sure in the accordion the ItemBody has the same semantic as children.

For the types, I know you are discussing with @divdavem

**Comment by quentinderoubaix** [comment]
PR was split into multiple ones due to the size.

**Pull request closed without merging**